components/batch-system/src/batch.rs

```diff
@@ -325,14 +327,32 @@ pub trait PollHandler<N, C> {
     }
 }
 
+type JoinHandleCollection = Arc<Mutex<HashMap<ThreadId, (Option<JoinHandle<()>>, bool)>>>;
```

Comment:
How about define a struct for `(Option<JoinHandle<()>>, bool)`>?

Reply:
struct tuple looks better, but I didn’t come up with a suitable name